docs/proposals/distributedstorage/distributedstorage.md

```diff
@@ -0,0 +1,594 @@
+
+#### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- **Resource Require** In order to configure the Ceph storage cluster, at least one of these local storage options are required:
+    - Raw devices (no partitions or formatted filesystems)
+    - Raw partitions (no formatted filesystem)
+    - LVM Logical Volumes (no formatted filesystem)
+    - Persistent Volumes available from a storage class in block mode
+
+#### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- **Version Compatibility** It is recommended that users use a newer version of Kubernetes, as older versions are not tested with Rook. Kubernetes v1.19 or higher is supported by Rook.
+
+### Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In this section, we will dive into the detailed API designs for the Unified Distributed Cloud Storage feature.
+
+These APIs are designed to facilitate Kurator's integration with Rook to enable the required functionality.
+
+In contrast to Rook, we may need to adapt the Unified Distributed Cloud Storage to reflect our new strategy and decisions.
+
+#### Unified Distributed Storage System API
+
+Here's the preliminary design for the Unified Distributed Storage System API:
+
+```console
+
+//  is the DistributedStorage schema for the DistributedStorage's API.
+type DistributedStorage struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              DistributedStorageSpec `json:"spec"`
+
+    // ClusterStatus is the current status of the distributed storage performed within this cluster
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	// +nullable
+    ClusterStatus rookv1.ClusterStatus `json:"status,omitempty"`
+}
+
+type DistributedStorageSpec struct {
+    // ClusterName is the Name of the cluster where the storage is being performed
+    // +optional
+    ClusterName string `json:"clusterName,omitempty"`
+
+    // ClusterKind is the kind of ClusterName recorded on Kurator
+    // +optional
+    ClusterKind string `json:"clusterKind,omitempty"`
+
+    // ClusterSpec indicates the state that the cluster wants to be in
+    // +optional
+    ClusterSpec StorageClusterSpec `json:"spec"`
```

Comment:
what if if multiple clusters use the same StorageClusterSpec? 
Is there a potential for duplicate StorageClusterSpec definitions if multiple clusters use the same spec? 
you can consider fleet cluster, see https://github.com/kurator-dev/kurator/blob/76fcb956d5585685a280f1f64306fce38722a858/pkg/fleet-manager/fleet_cluster.go#L45-L74

Reply:
Previously I thought fleet was able to filter managed clusters by label, but fleet doesn't support it. It will allow users to directly specify which clusters use the same ClusterSpec to avoid having multiple identical ClusterSpecs.